[EVAL] MultiChallenge

akshathmangudi opened the pull request:
## Overview
Resolves #1019 

Current status: READY FOR REVIEW

The current PR integrates [MultiChallenge](https://huggingface.co/datasets/nmayorga7/multichallenge), a difficult benchmark that tests the ability of models to handle multi-turn conversation with human users. The PR consists of a single file `multi_challenge.py`, which contains the general structure of including a prompt function, and loading it as a configuration with `LightEvalTaskConfig`.

The implementation was also tested with the command
```bash
lighteval eval "hf-inference-providers/openai/gpt-oss-20b" multi_challenge \
--bundle-dir multi-challenge --repo-id akshathmangudi/multi-challenge-gpt4o  \ 
--max-samples 10 --public
```

and results have been uploaded to my space [here](https://huggingface.co/spaces/akshathmangudi/multi-challenge-gpt4o).

NathanHB commented on src/lighteval/tasks/tasks/multi_challenge.py:
```diff
@@ -0,0 +1,168 @@
+
+
+@scorer(metrics=[accuracy(), stderr()])
+def multi_challenge_scorer():
```
great ! But it might be good to use the default inspect judge with the custom JUDGE_PROMPT instead ?

https://inspect.aisi.org.uk/reference/inspect_ai.scorer.html#model_graded_fact

akshathmangudi commented:
thanks for the feedback! 

i have addressed your comments. let me know if this is okay! 

p.s. i tried to run the evaluation task, but i ran out of credits 😅 
<img width="1225" height="228" alt="image" src="https://github.com/user-attachments/assets/364cc745-41d5-436d-b958-536dd2fe6ea5" />

NathanHB commented on src/lighteval/tasks/tasks/multi_challenge.py:
```diff
@@ -0,0 +1,177 @@
+            "pass_criteria": line["PASS_CRITERIA"],
+            "conversation": conversation,
+        },
+    )
```
we can remove this function as we do plan on supporting the bench for other than inspect-ai

akshathmangudi replied:
> we can remove this function as we do plan on supporting the bench for other than inspect-ai

@NathanHB, can you tell me what you mean by this? because the code above would support for backends other than inspect-ai right?

NathanHB commented on src/lighteval/tasks/tasks/multi_challenge.py:
```diff
@@ -0,0 +1,177 @@
+conversational, generation, instruction-following
+
+paper:
+https://arxiv.org/abs/2501.17399
```
```suggestion
https://arxiv.org/abs/2501.17399

starred:
true
```

akshathmangudi commented:
perfect, happy holidays!!

akshathmangudi commented:
> hey @akshathmangudi thanks for the changes ! I will have some days off for the end of year so will be reviewing / testing your PRs beginning of the year, have a nice holiday 🤗

hey @NathanHB, please let me know if there are any changes to be made. i would love to get this merged!

akshathmangudi commented:
absolutely, thanks for the help! 

looking forward to contribute more to the repo.
